docs/extensibility-http.md

```diff
@@ -0,0 +1,525 @@
+                        discoveryEndpoint:
+                          type: string
+...
+                  oneOf:
+                    - required: ["http"]
+```
+
+## Shared Configuration
+
+Revise `./shared/src/akri/configuration.rs`:
+
+```rust
+pub enum ProtocolHandler {
+    http(HTTPDiscoveryHandlerConfig),
+    ...
+}
+```
+
+And:
+
+```rust
+/// This defines the HTTP data stored in the Configuration
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct HTTPDiscoveryHandlerConfig {
+    pub discovery_endpoint: String,
+}
+```
+
+## Docker
+
+Revised `./dockerignore` to ensure `docker build ...` succeed:
+
+```console
+# Don't ignore these
+!samples/brokers/http
+!shared/
+```
+
+## Revise workspace Cargo file
+
+Revise `./Cargo.toml`:
+
+```TOML
+[workspace]
+members = [
+    ...
+    "samples/brokers/http",
+    ...
+]
+```
+
+> **NOTE ** This also ensures `http` inclusion using [rust-analyzer](https://github.com/rust-analyzer/rust-analyzer)
+
+## Revise workspace Cross file
+
+Revise `./Cross.toml`:
+
+```TOML
+[target.x86_64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:x86_64-unknown-linux-gnu-0.1.16-0.0.6"
+
+[target.armv7-unknown-linux-gnueabihf]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:armv7-unknown-linux-gnueabihf-0.1.16-0.0.6"
+
+[target.aarch64-unknown-linux-gnu]
+image = "ghcr.io/[[GITHUB-USER]]/rust-crossbuild:aarch64-unknown-linux-gnu-0.1.16-0.0.6"
+```
+
+## Build Akri Agent|Controller
+
+```bash
+USER=[[GTHUB-USER]]
+
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make rust-crossbuild
+
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-agent
+PREFIX=ghcr.io/${USER} BUILD_AMD64=1 BUILD_ARM32=0 BUILD_ARM64=0 make akri-controller
+```
+
+## Build|Push Broker (HTTP)
+
+```bash
+HOST="ghcr.io"
+USER=[[GITHUB-USER]]
+REPO="akri" # Or your preferred GHCR repo prefix
+TAGS="v1"
+
+docker build \
+--tag=${HOST}/${USER}/${REPO}:${TAGS} \
+--file=./samples/brokers/http/Dockerfiles/standlone \
+. && \
+docker push ${HOST}/${USER}/${REPO}-broker:${TAGS}
+```
+
+Revise `./samples/brokers/http/kubernetes/http.yaml` to reflect the image and the digest.
+
+You may manually confirm the image and digest using the output from the build or push commands, or:
+
+```bash
+IMAGE="$(docker inspect --format='{{index .RepoDigests 0}}' ${HOST}/${USER}/${REPO}-broker:${TAGS})"
+
+sed \
+--in-place \
+"s|IMAGE|${IMAGE}|g"
+./samples/brokers/http/kubernetes/http.yaml
+```
+
+> **NOTE** If you're using a non-public repo, you can create an `imagePullSecret` to authenticate
+
+## Confirm GitHub Packages
+
+You may confirm that the `agent`, `controller` and `http` images were push to GitHub Container Registry by browsing:
+
+https://github.com/[[GITHUB-USER]]?tab=packages
+
+
+## Deploy Device(s) & Discovery
+
+```bash
+cd ./samples/apps/http-apps
+
+HOST="ghcr.io"
+USER=[[GITHUB-USER]]
+REPO="http-apps" # Or your preferred GHCR repo prefix
```

Comment:
maybe `REPO="http-apps"` should be something like `IMAGE_PREFIX="http-apps"`

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/60620b61185473dab04cd2fb77a3344905a2825b